testing/mock/v2/ibc_app.go

```diff
@@ -9,7 +9,8 @@ import (
 )
 
 type IBCApp struct {
-	OnSendPacket    func(ctx context.Context, sourceID string, destinationID string, sequence uint64, data channeltypesv2.PacketData, signer sdk.AccAddress) error
-	OnRecvPacket    func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) channeltypesv2.RecvPacketResult
-	OnTimeoutPacket func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) error
+	OnSendPacket            func(ctx context.Context, sourceID string, destinationID string, sequence uint64, data channeltypesv2.PacketData, signer sdk.AccAddress) error
+	OnRecvPacket            func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) channeltypesv2.RecvPacketResult
+	OnTimeoutPacket         func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, relayer sdk.AccAddress) error
+	OnAcknowledgementPacket func(ctx context.Context, sourceID string, destinationID string, data channeltypesv2.PacketData, acknowldgement []byte, relayer sdk.AccAddress) error
```

Comment:
we could probably rename all `sourceID` and `destinationID` here but it can be done in a separate PR imho

Reply:
Ah I think actually Cian did it in his PR. I will take care of merge conflicts